src/plugins/vis_type_timelion/public/helpers/timelion_request_handler.ts

```diff
@@ -110,7 +121,9 @@ export function getTimelionRequestHandler({
             interval: visParams.interval,
             timezone,
           },
-          sessionId,
+          sessionId: searchSessionId,
+          isRestore: isCurrentSession() ? dataSearch.session.isRestore() : false,
```

Comment:
nit: what if we group all properties related to session together e.g.

```
{
  session: {
     sessionId: searchSessionId,
     isRestore, 
     isStored
  }
}
```
it seems to me that it will be more clear

Reply:
Improved